docs/src/tutorials/Nonlinear programs/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,312 @@
+
+# > #### Disclaimer
+# > Do not try to actually land a Space Shuttle using this notebook! 😛
+# > There's no mesh refinement going on, which can lead to unrealistic trajectories having position
+# > and velocity errors with orders of magnitude $10^4$ ft and $10^2$ ft/sec, respectively.
+
+using Interpolations
+using JuMP, Ipopt #, KNITRO
+
+## Global variables
+w  = 203000.0  # weight (lb)
+g₀ = 32.174    # acceleration (ft/sec^2)
+m  = w / g₀    # mass (slug)
+
+## Aerodynamic and atmospheric forces on the vehicle
+ρ₀ =  0.002378
+hᵣ =  23800.0
+Rₑ =  20902900.0
+μ  =  0.14076539e17
+S  =  2690.0
+a₀ = -0.20704
+a₁ =  0.029244
+b₀ =  0.07854
+b₁ = -0.61592e-2
+b₂ =  0.621408e-3
+c₀ =  1.0672181
+c₁ = -0.19213774e-1
+c₂ =  0.21286289e-3
+c₃ = -0.10117249e-5
+
+## Initial conditions
+hₛ = 2.6          # altitude (ft) / 1e5
+ϕₛ = deg2rad(0)   # longitude (rad)
+θₛ = deg2rad(0)   # latitude (rad)
+vₛ = 2.56         # velocity (ft/sec) / 1e4
+γₛ = deg2rad(-1)  # flight path angle (rad)
+ψₛ = deg2rad(90)  # azimuth (rad)
+αₛ = deg2rad(0)   # angle of attack (rad)
+βₛ = deg2rad(0)   # bank angle (rad)
+tₛ = 1.00         # time step (sec)
+
+## Final conditions, the so-called Terminal Area Energy Management (TAEM)
+hₜ = 0.8          # altitude (ft) / 1e5
+vₜ = 0.25         # velocity (ft/sec) / 1e4
+γₜ = deg2rad(-5)  # flight path angle (rad)
+
+## Number of mesh points (knots) to be used
+n = 2009
+
+## Integration scheme to be used for the dynamics
+integration_rule = "rectangular"
+nothing #hide
+
+#-
+
+user_options_ipopt = (
+    "mu_strategy" => "monotone",
+    "linear_solver" => "ma27",  # For the best results, it is advised to experiment different linear solvers.
+                                ## If Ipopt is not compiled with MA27/MA57, it may fallback to 'MUMPS'.
+                                ## In general, the linear solver MA27 is much faster than MUMPS.
+)
+
+## user_options_knitro = (
+##     "algorithm" => KNITRO.KN_ALG_BAR_DIRECT,
+##     "bar_murule" => KNITRO.KN_BAR_MURULE_QUALITY,
+##     "linsolver" => KNITRO.KN_LINSOLVER_MA27,
+## )
+
+## Create JuMP model, using Ipopt as the solver
+model = Model(optimizer_with_attributes(Ipopt.Optimizer, user_options_ipopt...))
+## model = Model(optimizer_with_attributes(KNITRO.Optimizer, user_options_knitro...))
+
+@variables(model, begin
+               0 ≤ scaled_h[1:n]                # altitude (ft) / 1e5
+                          ϕ[1:n]                # longitude (rad)
+    deg2rad(-89) ≤        θ[1:n] ≤ deg2rad(89)  # latitude (rad)
+            1e-4 ≤ scaled_v[1:n]                # velocity (ft/sec) / 1e4
+    deg2rad(-89) ≤        γ[1:n] ≤ deg2rad(89)  # flight path angle (rad)
+                          ψ[1:n]                # azimuth (rad)
+    deg2rad(-90) ≤        α[1:n] ≤ deg2rad(90)  # angle of attack (rad)
+    deg2rad(-89) ≤        β[1:n] ≤ deg2rad( 1)  # bank angle (rad)
+    ##        0.5 ≤       Δt[1:n] ≤ 1.5          # time step (sec)
```

Comment:
Unused?

Reply:
Can you have a look at this https://github.com/jump-dev/JuMPTutorials.jl/pull/97#discussion_r572812250? Let me know if you think I should remove the commented line, or if I should add more explanation to it.

Reply:
Perhaps add an `!!! info` block after the variable definitions with the explanation?

In general, we should remove commented code, because new users are just going to say "why is this here?"